src/main/webapp/feed.js

```diff
@@ -410,37 +410,52 @@ function showChallengeCompletePage(challenge, newCompletion) {
 
     newCompletion ? showNewChallengeCompletePage(challenge) : showOldChallengeCompletePage(challenge);
 
-    setPrevButton(challenge.get("steps").length+1, challenge);
+    setPrevButton(challenge.steps.length+1, challenge);
 }
 
 async function showNewChallengeCompletePage(challenge) {
     const text = document.getElementById("challenge-complete-text");
-    let newChallengeId = findNextUncompletedChallenge(challenge.get("id"));
+    let newChallengeId = findNextUncompletedChallenge(challenge.id);
     if (newChallengeId == -1) {
         text.innerHTML = "All challenges complete!";
+        await getServerChallenges(3);
     }
     else {
-        text.innerHTML = `${challenge.get("title")} challenge complete!<br>Next up is the <b>${challenges[newChallengeId].get("title")}</b> challenge`;
+        text.innerHTML = `${challenge.challenge_type} challenge complete!<br>Next up is the <b>${challengeMap.get(newChallengeId).name}</b> challenge`;
     }
+
+    //instead of newChallengeID us key name from Challenge Data
     let idToken = getIdToken();
     const putRequest = new Request(`/user?id_token=${idToken}&chal=${newChallengeId}`, {method: 'PUT'});
+
     user = await fetch(putRequest).then(response => response.json());
 }
 
+/*async function setnewChallenges(){
+    fetch('/challenges?num-challenges=' + 3);
+
+}
+ */ 
+async function sendcompletedChallenges() {
+    fetch('/challenges', {methdo:'POST'});
+  
+}
+
 function findNextUncompletedChallenge(prevChallengeId) {
-    for (i = 1; i < challenges.length; i++) {
-        let id = (prevChallengeId+i) % challenges.length;
-        let status = user.challenge_statuses[id];
-        if (status < challenges[id].get("steps").length) {
-            return id;
+    for(chalId of challengeMap.keys()) {
+      if(chalId != prevChallengeId) {
+        let status = user.challenge_statuses[chalId];
+        if (status < challengeMap.get(chalId).steps.length) {
+          return chalId;
```

Comment:
Are you requiring the user to do the challenges in order?

Reply:
No, I do not. I believe this will have the same functionally as how you previously wrote it. I just had to adapt it to the map.